src/traces/bar/cross_trace_calc.js

```diff
@@ -119,10 +123,44 @@ function setGroupPositions(gd, pa, sa, calcTraces, opts) {
             }
             break;
     }
-
     collectExtents(calcTraces, pa);
 }
 
+function standardizeCornerradius(calcTraces) {
+    if(calcTraces.length < 2) return;
+    var i, calcTrace, fullTrace;
+    var cr;
+    for(i = 0; i < calcTraces.length; i++) {
+        calcTrace = calcTraces[i];
+        fullTrace = calcTrace[0].trace;
+
+        fullTrace.marker = fullTrace.marker || {};
+        cr = fullTrace.marker.cornerradius;
+        if(cr !== undefined) break;
+    }
+    // If any trace has cornerradius, set all other traces
+    // to match first trace which has cornerradius
+    if(cr !== undefined) {
+        for(i = 0; i < calcTraces.length; i++) {
+            calcTrace = calcTraces[i];
+            fullTrace = calcTrace[0].trace;
+
+            fullTrace.marker = fullTrace.marker || {};
+            fullTrace.marker.cornerradius = cr;
+        }
+    // Otherwise, remove cornerradius from all other traces
+    } else {
+        for(i = 0; i < calcTraces.length; i++) {
+            calcTrace = calcTraces[i];
+            fullTrace = calcTrace[0].trace;
+
+            if(fullTrace.marker && fullTrace.marker.cornerradius) {
+                delete fullTrace.marker.cornerradius;
+            }
+        }
+    }
```

Comment:
It's a little bit subtle - and it's conceivable this isn't a bug but it's hard to be sure given all the possible ways of editing a figure - but this is an antipattern: `fullTrace` should not be modified after the `supplyDefaults` step is done, and this is the `calc` step. More precisely the regular attributes in it should be left as they were set by `supplyDefaults`, sometimes we've altered private attributes like `fullTrace._extremes`.

But this is exactly what the calc step is for anyway, and since what you're calculating is a trace-wide property I'd suggest putting it in `calcTrace[0].t` - the same value for every trace in the stack - then pull that value back out both in the plot and the legend style steps. Furthermore you can use this opportunity to calculate efficiently which form it has (px or %) and save both the form and the numeric value (as a plain number) in `calcTrace[0].t`.

Reply:
Makes sense. Updated!